scripts/metadata_table.py

```diff
@@ -96,475 +125,1137 @@
                         self._dependencies += [ v.strip() for v in value.rstrip(",").split(",") ]
                 elif key == 'relative_path':
                     relative_path_local = value.strip()
+                elif key == 'process':
+                    self.__process_type = value
                 else:
-                    raise ParseSyntaxError("metadata table start property",
-                                           token=value, context=self._pobj)
-                # End if
-            # End for
-            curr_line, curr_line_num = self._pobj.next_line()
-        # End while
+                    self.__pobj.add_syntax_err("metadata table start property",
+                                               token=value)
+                    self.__process_type = 'INVALID' # Allow error continue
+                    self.__section_valid = False
+                # end if
+            # end for
+            curr_line, _ = self.__pobj.next_line()
+        # end while
         if self.title is None:
-            raise ParseSyntaxError("metadata header start, no table name",
-                                   token=curr_line, context=self._pobj)
-        elif self.header_type is None:
-            raise ParseSyntaxError("metadata header start, no table type",
-                                   token=curr_line, context=self._pobj)
-        elif self.header_type == "ddt":
-            register_fortran_ddt_name(self.title)
-        # End if
-        # Add relative path to dependencies
-        if self.dependencies and relative_path_local:
-            self._dependencies = [ os.path.join(relative_path_local, v) for v in self.dependencies]
+            self.__pobj.add_syntax_err("metadata header start, no table name",
+                                       token=curr_line)
+            self.__section_valid = False
+        # end if
+        if self.header_type is None:
+            self.__pobj.add_syntax_err("metadata header start, no table type",
+                                       token=curr_line)
+            self.__section_valid = False
+        # end if
+        if ((self.header_type != SCHEME_HEADER_TYPE) and
+            (self.process_type != UNKNOWN_PROCESS_TYPE)):
+            emsg = "process keyword only allowed for a scheme"
+            self.__pobj.add_syntax_err(emsg, token=curr_line)
+            self.__process_type = UNKNOWN_PROCESS_TYPE # Allow error continue
+            self.__section_valid = False
+        # end if
+        mismatch = self.section_table_mismatch(table_name, table_type)
+        if mismatch:
+            self.__pobj.add_syntax_err(mismatch)
+            self.__section_valid = False
+        # end if
+        if logger:
+            logger.info("Parsing {} {}{}".format(self.header_type,
+                                                 self.title, start_ctx))
+        # end if
+        if self.header_type == "ddt":
+            known_ddts.append(self.title)
+        # end if
```

Comment:
There should be code here which is missing. What happened?

Reply:
I am not sure, there were several merge conflicts that I had to resolve. What is missing here?

Reply:
Do a diff against the feature/capgen version. After you merge #395, there should be no difference.
For this PR to be okay, there needs to be only one change between capgen code in feature/capgen and the code in this PR. Currently, that is not true.